src/parser/func_type.cpp

```diff
@@ -1470,131 +1471,1614 @@ pt_are_equivalent_types (const PT_ARG_TYPE def_type, const PT_TYPE_ENUM op_type)
   return false;
 }
 
+// TODO: remove me
+#define PT_COLL_WRAP_TYPE_FOR_MAYBE(type) \
+  ((PT_IS_CHAR_STRING_TYPE (type)) ? (type) : PT_TYPE_VARCHAR)
+
+#define PT_IS_CAST_MAYBE(node) \
+  ((node)->node_type == PT_EXPR && (node)->info.expr.op == PT_CAST \
+   && (node)->info.expr.arg1 != NULL && (node)->info.expr.arg1->type_enum == PT_TYPE_MAYBE)
+
+static PT_NODE *pt_check_function_collation (PARSER_CONTEXT *parser, PT_NODE *node);
+static int pt_character_length_for_node (PT_NODE *node, const PT_TYPE_ENUM coerce_type);
+
 /*
- * pt_get_equivalent_type () - get the type to which a node should be
- *			       converted to in order to match an expression
- *			       definition
- *   return	  : the new type
- *   def_type(in) : the type defined in the expression signature
- *   arg_type(in) : the type of the received expression argument
+ * pt_eval_function_type () -
```

Comment:
The comment is not matched with the real function name "pt_eval_function_type_aggregate"